src/body/body.rs

```diff
@@ -382,7 +382,7 @@ impl HttpBody for Body {
                 ..
             } => match ready!(Pin::new(trailers_rx).poll(cx)) {
                 Ok(t) => Poll::Ready(Ok(Some(t))),
-                Err(_) => Poll::Ready(Err(crate::Error::new_closed())),
+                Err(_) => Poll::Ready(Ok(None)),
```

Comment:
@seanmonstar If we don't send any trailers, polling the oneshot receiver returns `Err(Canceled)` so if we return `Err(..)` in this case, receive process will result in error. That's why I changed this to `Ok(None)`. This way we don't have to send empty trailers. Is it ok though?

Reply:
Oh yes, this change is the correct behavior. Otherwise, when people upgrade hyper then suddenly in H2 when the trailers are polled they'd error the stream since they weren't sending any before.